### PR TITLE
Use cursor pagination with many operations

### PR DESCRIPTION
**Pull request by SferaDev**
Closes: #314 

- Change internal methods to use cursor navigation instead of offset navigation
- Behaviour change: Ignore ``sorting/filtering`` when sending cursor navigation.
  - Backend would fail with that request
  - In userland TypeScript type guards do not allow it
  - Our internal methods could still generate bad queries
  - Ignore them for our internal methods and JavaScript users

Regarding this behaviour change, I have mixed feelings to be honest. I kind of like that we do not error out on impossible situations, but could mislead a JavaScript user who doesn't have type support.

### CODE REVIEW

File: test/integration.test.ts [hunk 1]
```diff
@@ -266,7 +265,7 @@ describe('integration tests', () => {
     expect(page2And3.records).toEqual([...page2.records, ...page3.records]);
   });
 
-  test('fails if sending cursor with sorting', async () => {
+  test('we ignore sorting if sending it with cursor', async () => {
```
Comment (SferaDev) [inline test/integration.test.ts:268]: This is the behaviour change we are introducing for JS users

**Review by gimenete — COMMENTED** [review]
> Regarding this behaviour change, I have mixed feelings to be honest. I kind of like that we do not error out on impossible situations, but could mislead a JavaScript user who doesn't have type support.

Yeah. I'd prefer to throw than having unexpected behavior that is hard to debug.

The cursor pagination changes LGTM.

**Comment by SferaDev** [comment]
> Yeah. I'd prefer to throw than having unexpected behavior that is hard to debug.

Fixed!

**Review by gimenete — APPROVED** [review]
🚀

**Pull request merged**